packages/web/src/components/EditorLayout/index.jsx

```diff
@@ -126,7 +114,7 @@ export default function EditorLayout() {
           <Button
             variant="contained"
             size="small"
-            onClick={() => onFlowStatusUpdate(!flow.active)}
+            onClick={() => onFlowStatusUpdate(flow ? !flow.active : false)}
```

Comment:
We should disable this button if the `flow` doesn't exist. Due to the hardcoded `false` value here, this iteration may make a rare situation possible.

- open editor with an already published flow 
- `flowId` comes from the URL
- `flow` is yet to be loaded
- this button shows "publish flow" text as `flow.active` is not loaded yet
- user takes click action on this button
- `onFlowStatusUpdate(false)` takes place in the absence of `flow` and unpublishes the published flow, but the user intended to publish it due to the button text.

Reply:
We should also disable it when the user doesn't have `flow.publish` permission in their role.